### ISSUE TITLE
BarGraphItem is not showing if bar coordinates are huge

### ISSUE THREAD

**Issue by noonchen** [pyqtgraph/pyqtgraph#2550]
### Short description
See example below, use `data` to create a `BarGraphItem`, there will be nothing, whereas use `data_ok` everything is normal.

### Code to reproduce
```python
"""
Simple example using BarGraphItem
"""

import numpy as np
import pyqtgraph as pg

win = pg.plot()
win.setWindowTitle('pyqtgraph example: BarGraphItem')

# prepare data

# bars are invisible, might show up when resizing
data = np.random.randint(2E9, 4E9, 1000)

# # if data is not very big, everything is fine
# data_ok = np.random.randint(200, 400, 1000)

counts, edges = np.histogram(data, bins=50)
height = edges[1]-edges[0]
bg1 = pg.BarGraphItem(x0=0, y=edges[:-1], height=height, width=counts, brush='r')
# set viewbox range
vb = win.getPlotItem().getViewBox()
vb.setLimits(xMin=0, 
             xMax=counts.max() * 1.2,
             yMin=edges[0] * 0.9, 
             yMax=edges[-1] * 1.1)

win.addItem(bg1)

if __name__ == '__main__':
    pg.exec()
```

### Real behavior
Blank scene.
<img width="752" alt="image" src="https://user-images.githubusercontent.com/26359487/206895825-8e77897a-3b3f-4b19-83b4-cfd16a7c66e1.png">

### Tested environment(s)

 * PyQtGraph version: 0.13.1
 * Qt Python binding: PyQt5 5.15.4 Qt 5.15.2
 * Python version: 3.9
 * NumPy version: 1.20.3
 * Operating system: macOS
 * Installation method: pip

**Comment by noonchen** [comment]
FYI, if we right click on this blank scene and export it to png image, we can get the expected result.

It looks like something's wrong in viewbox?

**Comment by pijyoi** [comment]
The issue seems to be that `BarGraphItem`, `NonUniformImage`, `PColorMeshItem` and `customGraphicsItem` all compute their `boundingRect` using `QPicture.boundingRect() -> QRect`. As `QPicture`'s bounding rectangle is only a `QRect` and not a `QRectF`, its limits are determined by the range of an `int32`.

For `BarGraphItem`, it seems that your issue could be solved if `BarGraphItem::boundingRect()` were defined as:
```python
def boundingRect(self):
    return self.shape().boundingRect()
```
where `self.shape()` is a `QPainterPath` and `QPainterPath.boundingRect() -> QRectF`. 


By the way, `data = np.random.randint(2E9, 4E9, 1000)`  raises an exception on Windows platform

**Comment by noonchen** [comment]
Thanks @pijyoi , It works!

You could close this issue now.

**Comment by j9ac9k** [comment]
@pijyoi we should probably implement the suggested feature in the library, _but_ `self.shape().boundingRect()` is much more expensive than `QPicture.boundingRect()` if I remember right.  Can you think of a potentially easier method to check if calling self.shape().boundingRect() would be needed?

**Comment by pijyoi** [comment]
Each bar is drawn as a rectangle. It should be possible to build up the bounding rectangle while drawing each bar.

**Comment by pijyoi** [comment]
There's also `self.shape().controlPointRect()` which "_is significantly faster to compute than the exact boundingRect()_".
For both `QPainterPath::boundingRect()` and `QPainterPath::controlPointRect()`, the results are cached.

**Comment by j9ac9k** [comment]
Re-opening as we've reverted #2554 with #2560

**Comment by j9ac9k** [comment]
Closing as I think it's fixed with #2565
